model/src/form/form-metadata/types.ts

```diff
@@ -1,5 +1,20 @@
 import { formMetadataSchema } from '~/src/form/form-metadata/index.js'
 
+/**
+ * Interface for metadata state (draft & live)
+ */
+export interface FormMetadataState {
+  /**
+   * The date the form state was created
+   */
+  createdAt: Date
```

Comment:
Won't these need to be strings at some point? e.g. ISO-8601 dates

So we can correctly "type" what comes in/out of storage

```
createdAt: '2024-04-22T20:47:05.335-14:00'
```

Reply:
They come out of storage as `Date`s (mongo uses bson which has support for dates, not json).

The only time they are stringified is when they're over the wire from the forms-manager API endpoint. Running them through the Joi schema like I do [here](https://github.com/DEFRA/forms-runner/blob/1ff8c3a815125309eae405e405d63feddbc752eb/src/server/plugins/engine/services/formsService.js#L16) gets them back to `Date`s.